hack/go-mod-env.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+# If inside GOPATH/src, set GO111MODULE=on
+setGoModEnv() {
+  pwdPath=$(pwd -P 2>/dev/null || env PWD= pwd)
+  goPath=$(go env GOPATH)
+  cd "$goPath" || exit
+  goPath=$(pwd -P 2>/dev/null || env PWD= pwd)
+  cd "$pwdPath" || exit
+  if [ "${pwdPath#"$goPath"}" != "${pwdPath}" ]; then
+    export GO111MODULE=on
```

Comment:
Is this export necessary?

Reply:
It is necessary when [invoking the `go` command in a directory inside of the `$GOPATH/src`](https://github.com/golang/go/wiki/Modules#how-to-install-and-activate-module-support).

We won't need it anymore once we upgrade to go 1.13:

> GO111MODULE=auto enables module-mode if any go.mod is found, even inside GOPATH. (Prior to Go 1.13, GO111MODULE=auto would never enable module-mode inside GOPATH).